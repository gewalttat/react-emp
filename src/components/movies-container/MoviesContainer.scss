.movies-container {
    margin-top: 10px;
    padding-top: 20px;
    padding-left: 8%;
    background: #232323;
}

.movies-grid {
    margin-top: 70px;
}