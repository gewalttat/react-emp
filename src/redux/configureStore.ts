/* eslint-disable @typescript-eslint/ban-ts-comment */
import rootReducer, { rootSaga } from './rootReducer';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import createSagaMiddleware, { END } from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

const createStore = () => {
  const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware),
  });

  // @ts-ignore
  store.runSaga = () => sagaMiddleware.run(rootSaga);
  // @ts-ignore
  store.close = () => store.dispatch(END);

  return store;
};

export default createStore;