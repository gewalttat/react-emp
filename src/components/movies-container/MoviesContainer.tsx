import { Grid, Typography } from '@material-ui/core';
import React, { FC, useEffect, useState } from 'react';
import ErrorBoundary from '../error-boundary/ErrorBoundary';
import { MovieCard } from '../movie-card/MovieCard';
import { SortingFilter } from '../sorting-filter/SortingFilter';
import './MoviesContainer.scss';
import { useDispatch, useSelector } from 'react-redux';
import { getMovies, filterMovies, filterAndSortMovies, sortMovies, selectMovies } from '../../redux/moviesReducer';

export interface MovieData {
  budget?: number;
  genres?: string[];
  id?: number;
  overview?: string;
  poster_path?: string;
  release_date?: string | null;
  revenue?: number;
  runtime?: number;
  tagline?: string;
  title?: string;
  vote_average?: number;
  vote_count?: number;
}

export interface ResponseMoviesType {
  data: MovieData[];
  limit: number;
  offset: number;
  totalAmount: number;
}

export const MoviesContainer: FC = () => {
  const [filteredMovies, setFilteredMovies] = useState<MovieData[]>();
  const [filter, filterChanged] = useState<string>();
  const [sortBy, sortByChanged] = useState<string>();
  const { movies } = useSelector(selectMovies);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(
      filter && !sortBy ? filterMovies(filter.toLowerCase())
        : sortBy && !filter ? sortMovies(sortBy.toLowerCase())
          : filter && sortBy ? filterAndSortMovies({ sortBy: sortBy.toLowerCase(), filter: filter.toLowerCase() })
            : getMovies());
  }, [filter, sortBy]);

  useEffect(() => {
    setFilteredMovies(() => movies);
  }, [movies]);

  return (
    <div className='movies-container'>
      <ErrorBoundary>
        <SortingFilter filterChanged={filterChanged} sortByChanged={sortByChanged} />
        <hr style={{
          width: '90%',
          marginLeft: '0px',
          marginTop: '0px',
          borderColor: '#424242'
        }} />
        <Typography
          sx={{
            color: '#fff',
            fontFamily: 'Montserrat',
            fontWeight: 500,
            fontSize: '20px',
            lineHeight: '24px',
            paddingTop: '30px'
          }}
          variant="body2"
          color="text.secondary">
          <b>{filteredMovies?.length}</b>movies found
        </Typography>
        <div className='movies-grid'>
          <Grid
            container
            spacing={{
              xs: 2,
              md: 3
            }}
            columns={{
              xs: 4,
              sm: 8,
              md: 12.5
            }}>
            {filteredMovies?.map((i, index) => (
              <Grid
                item
                xs={2}
                sm={4}
                md={4}
                key={index}>
                <MovieCard movieData={i} />
              </Grid>
            ))}
          </Grid>
        </div>
      </ErrorBoundary>
    </div>
  )
}