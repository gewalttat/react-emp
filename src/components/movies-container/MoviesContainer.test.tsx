import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import {MoviesContainer} from './MoviesContainer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import {createMemoryHistory} from 'history'

const mockUseLocationValue = {
    pathname: "/testroute",
    search: '',
    hash: '',
    state: null
}
jest.mock('react-router', () => ({
    ...jest.requireActual("react-router") as Record<string, never>,
    useLocation: jest.fn().mockImplementation(() => {
        return mockUseLocationValue;
    })
}));

describe('With React Testing Library', () => {
  const initialState = {movies: []}
  const mockStore = configureStore();

  it('input renders correctly', () => {
    const store = mockStore(initialState);
    const tree = renderer.create(<Provider store={store}><MoviesContainer/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})