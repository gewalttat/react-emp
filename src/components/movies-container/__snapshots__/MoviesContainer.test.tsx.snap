// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library input renders correctly 1`] = `
<div
  className="movies-container"
>
  <div
    className="sorting-filter-container"
  >
    <div
      className="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": -0,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="secondary tabs example"
          aria-orientation={null}
          className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        />
      </div>
    </div>
    <span
      className="sort-label"
    >
      sort by
    </span>
    <div
      className="filter-wrapper"
    >
      <div
        className="MuiFormControl-root css-1h8wmp1-MuiFormControl-root"
      >
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-vb4nm0-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-standard-label demo-simple-select-standard"
            className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
            id="demo-simple-select-standard"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="notranslate"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <hr
    style={
      Object {
        "borderColor": "#424242",
        "marginLeft": "0px",
        "marginTop": "0px",
        "width": "90%",
      }
    }
  />
  <p
    className="MuiTypography-root MuiTypography-body2 css-d646ni-MuiTypography-root"
  >
    <b />
    movies found
  </p>
  <div
    className="movies-grid"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-[object Object] css-1xgjpwf-MuiGrid-root"
    />
  </div>
</div>
`;
